### PR TITLE
Try to implement discard sequence #_ for deserialize

### PR DESCRIPTION
**Pull request by jollm**
See https://github.com/naomijub/edn-rs/issues/94

Noticed your issue post after PR #93
Hope it's okay I tried to implement this one, seemed maybe fun to try

### CODE REVIEW

**Comment by naomijub** [comment]
Loved it! Will check as soon as I am free

File: src/deserialize/parse.rs [hunk 1]
```diff
@@ -907,6 +926,96 @@ mod test {
...
+            .chars()
+            .enumerate();
+        let res = parse(edn.next(), &mut edn).unwrap();
+
+        assert_eq!(res, Edn::Empty)
+    }
+
+    #[test]
+    fn parse_discard_repeat_not_empty() {
+        let mut edn = "#_ ,, #_{discard again} #_ {:and :again} :okay {:a map}"
```
Comment (naomijub) [inline src/deserialize/parse.rs:993]: this seems weird, parsing `:okay {:a map}` seems like a problem to return:
```
Edn::Map(Map::new(
  map! {":a".to_string() => Edn::Symbol("map".to_string())}
))
```

Did you check the standard response? I would expect the lib to return the keyword :okay and ignore the map

File: src/deserialize/parse.rs [hunk 2]
```diff
@@ -907,6 +926,96 @@ mod test {
         )
     }
 
+    #[test]
```
Comment (naomijub) [inline src/deserialize/parse.rs:929]: I would say it is missing a test where the discard element is a sequence inside a sequence, like:
```
#_"random comment" [:a :b :c #_(:hello :world) :d]
``` 
then the result would be a vector with `[:a :b :c :d]`

File: src/deserialize/parse.rs (hunk 1 above)
Comment (jollm) [reply src/deserialize/parse.rs:993]: yea let me check what standard lib does with that.

this is how it works here: 
```
(discard 
  (discard {discard again}
           return (discard {:and :again}
                           return :okay)))
  return {:a map})
```

Comment (jollm) [reply src/deserialize/parse.rs:993]: Here it is:
```test.clj
clj-scratch.core> (edn/read-string "#_ ,, #_{discard again} #_ {:and :again} :okay {:a map}")
{:a map}
clj-scratch.core> 
```

**Review by naomijub — APPROVED** [review]

File: src/deserialize/parse.rs (hunk 1 above)
Comment (naomijub) [reply src/deserialize/parse.rs:993]: Nice

**Comment by jollm** [comment]
Wait, i found one that breaks:
```test.clj
clj-scratch.core> (edn/read-string "[:foo #_ foo]")
[:foo]
clj-scratch.core> 
```

Have a fix ready..

**Comment by naomijub** [comment]
I think you will need to rebase or create a new PR

**Pull request merged**